### PR TITLE
Add StructureMapUtilities Unit Tests

### LINKED ISSUE #1527: [Quest] - Add/Fix failing test for StructureMaps on Quest
**Name of feature to enhance**
1. Add test for StructureMaps and generate JSON StructureMaps
2. Fix failing test
 
**Description of feature**
1. Add test to aid in StructureMaps authoring

### PR DESCRIPTION
**Pull request by f-odhiambo**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1527

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by ndegwamartin** [comment]
@f-odhiambo could you run the `spotlessApply` command on this and push?

**Comment by f-odhiambo** [comment]
Done

File: android/quest/src/main/assets/configs/app/translations/strings_config.properties [hunk 1]
```diff
@@ -16,4 +16,5 @@ add.new.household=Add New Household
 reports=Reports
 transfer.data=Transfer Data
 profile=Profile
-add.family=Add Family
\ No newline at end of file
+add.family=Add Family
+maps=Map
```
Comment (ndegwamartin) [inline android/quest/src/main/assets/configs/app/translations/strings_config.properties:20]: Since the value is _map_(singular) the key should also match

**Review by pld — CHANGES_REQUESTED** [review]

File: android/quest/src/quest/res/values/test.xml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ndegwamartin) [inline android/quest/src/quest/res/values/test.xml:1]: We can remove this file if not in use

Comment (f-odhiambo) [reply android/quest/src/quest/res/values/test.xml:1]: Done

File: android/quest/src/main/assets/configs/app/translations/strings_config.properties (hunk 1 above)
Comment (f-odhiambo) [reply android/quest/src/main/assets/configs/app/translations/strings_config.properties:20]: Done

File: android/quest/src/test/java/org/smartregister/fhircore/quest/StructureMapTestUtilities.kt [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+                QuestionnaireResponse::class.java,
+                immunizationQuestionnaireResponseString
+            )
+
+        scu.transform(contextR4, baseElement, map, targetResource)
+
+        System.out.println(iParser.encodeResourceToString(targetResource))
+    }
+
+//    @Test
```
[the remark this answers is not in the document]
Comment (f-odhiambo) [reply android/quest/src/test/java/org/smartregister/fhircore/quest/StructureMapTestUtilities.kt]: Done

**Review by pld — APPROVED** [review]

**Review by pld — COMMENTED** [review]
why do we need to add these 3 files?

android/quest/src/maps/assets/measure/ANCIND01-bundle.json
android/quest/src/maps/res/drawable/ic_app_logo.xml
android/quest/src/maps/res/drawable/ic_launcher.xml

**Review by pld — COMMENTED** [review]
is that how we define a flavor? is that related to the map flavor? if so, I think you should have the code related to mapping in a separate PR

**Comment by f-odhiambo** [comment]
> is that how we define a flavour? is that related to the map flavour? if so, I think you should have the code related to mapping in a separate PR

Yes, my bad just pushed everything  @pld This is accurate. I was trying to add a new map config but yes will separate this.

**Comment by pld** [comment]
No worries, ok that sounds good, lmk when it’s split, this PR should be ready to go then, thank you

**Comment by f-odhiambo** [comment]
This PR is ready for merge CC @ndegwamartin @pld

**Review by pld — APPROVED** [review]

**Pull request merged**